src/main/java/edu/umich/ctools/esb/utils/WAPI.java

```diff
@@ -0,0 +1,213 @@
+package edu.umich.ctools.esb.utils;
+
+import java.util.HashMap;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.codec.binary.Base64;
+import org.json.JSONObject;
+
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.JsonNode;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+
+public class WAPI 
+{
+	private static Log M_log = LogFactory.getLog(WAPI.class);
+
+	private final int HTTP_SUCCESS = 200;
+	private final int HTTP_BAD_REQUEST = 400;
+	private final int HTTP_UNAUTHORIZED = 401;
+	private final int HTTP_FORBIDDEN = 403;
+	private final int HTTP_NOT_FOUND = 404;
+	private final int HTTP_GATEWAY_TMEOUT = 504;
+	private final int HTTP_UNKNOWN_ERROR = 666;
+
+	private final static String CONTENT_TYPE = "Content-Type";
+	private final static String CONTENT_TYPE_PARAMETER = "application/x-www-form-urlencoded";
+	private final static String AUTHORIZATION = "Authorization";
+	private final static String GRANT_TYPE = "grant_type";
+	private final static String CLIENT_CREDENTIALS = "client_credentials";
+	private final static String SCOPE = "scope";
+	private final static String PRODUCTION = "PRODUCTION";
+	private final static String ERROR_MSG = "No results due to error. See meta for more details.";
+
+	private final String BEARER = "Bearer";
+
+	private String apiPrefix;
+	private String tokenServer;
+	private String key;
+	private String secret;
+	private String renewal;
+	private String token;
+	
+	public WAPI() {
+		super();
+	}
+
+	//WAPI constructor will have single variable which will be a map holding all necessary variables
+	//The values for the map should come from a properties file used by the application that depends
+	//on this library. For example, if ccm uses the esb-utils library, the ccm properties should
+	//contain the values for the map being passed in to the WAPI constructor.
+	public WAPI(HashMap<String, String> value) {
```

Comment:
In my humble opinion, I don't think we need to put in the comment where should the key,secret,token of the application using ESBUtils should be coming from. I don't think project reference (like ccm) is needed.